goss-testing/scripts/data-validator.py

```diff
@@ -192,103 +206,96 @@ def are_hosts_sane(data, desired_keys):
-
-
-  for blob in node_class:
-    if blob['valid'] == False:
-      print_err(f"WARN: {blob['hostname']}: has boot params that differ from at least one other node\n{blob['params-orig']}\n")
-      err = 1
-
-  print_err("------------------------------------------------------------")
-
-  if err == 1:
-    return err
-
-
 def boot_params(data):
-  """Attempt to validate boot parameters from BSS
+  """Confirm that params required for booting are
+     present.
   """
-  worker = []
-  storage = []
-  management = []
-  err = 0
 
-  macreg = re.compile(r':([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}')
-  hostreg = re.compile(r'hostname=[a-zA-Z]*\-[a-zA-Z0-9]*')
-  httpreg = re.compile(r's=http://.*/')
+  err = 0
+  ncn_params = [
+	"biosdevname=1",
+	"console=tty0",
+	"console=ttyS0,115200",
+	"crashkernel=[0-9]*M",
+	"ds=",
+	"hostname=",
+	"ifname=mgmt0",
+	"ifname=mgmt1",
+	"initrd=initrd.img.xz",
+	"iommu=pt",
+	"log_buf_len=1",
+	"metal.server=",
+	"pcie_ports=native",
+	"psi=1",
+	"rd.auto=1",
+	"rd.bootif=0",
+	"rd.dm=0",
+	"rd.live.overlay.overlayfs=1",
+	"rd.live.overlay.thin=1",
+	"rd.live.overlay=LABEL=ROOTRAID",
+	"rd.live.ram=0",
+	"rd.live.squashimg=",
+	"rd.luks.crypttab=0",
+	"rd.lvm.conf=0",
+	"rd.lvm=1",
+	"rd.md.conf=1",
+	"rd.md.waitclean=1",
+	"rd.md=1",
+	"rd.multipath=0",
+	"rd.neednet=0",
+	"rd.net.dhcp.retry=5",
+	"rd.net.timeout.carrier=120",
+	"rd.net.timeout.iflink=120",
+	"rd.net.timeout.ifup=120",
+	"rd.net.timeout.ipv6auto=0",
+	"rd.net.timeout.ipv6dad=0",
+	"rd.peerdns=0",
+	"rd.retry=10",
+	"rd.shell",
+	"rd.skipfsck",
+	"rd.writable.fsimg=0",
+	"root=live:LABEL=SQFSRAID",
+	"rootfallback=LABEL=BOOTRAID",
+	"transparent_hugepage=never"
+	]
+
+  worker_params     = [ "rd.luks=0\s+" ]
+  storage_params    = [ "rd.luks\s+" ]
+  management_params = [ "rd.luks\s+" ]
 
   for blob in data:
     if "params" in blob and blob['cloud-init']['user-data'] is not None:
       hostname = blob['cloud-init']['user-data']['hostname']
-      params_stripd = macreg.sub("", blob['params'])
-      params_stripd = hostreg.sub("", params_stripd)
-      params_stripd = httpreg.sub("", params_stripd)
-      params_length = len("".join(params_stripd.split()))
-      node = {'hostname': hostname, 'params-orig': blob['params'], 'params-stripd': params_stripd, 'params-length': params_length}
-
-      if re.search("ncn-m0", hostname):
-        management.append(node)
+
       if re.search("ncn-w0", hostname):
-        worker.append(node)
-      if re.search("ncn-s0", hostname):
-        storage.append(node)
+        mod_params = ncn_params
+        mod_params = mod_params + worker_params
 
-  if score_params(worker): err = 1
-  if score_params(storage): err = 1
-  if score_params(management): err = 1
-  return err
+      elif re.search("ncn-s0", hostname):
+        mod_params = ncn_params
+        mod_params = mod_params + storage_params
+
+      elif re.search("ncn-m0", hostname):
```

Comment:
Do we want the `0` in both of these? It's unlikely we'll have a system with 100 storage or master nodes, but it might still be worth fixing this.